### PR TITLE
force charset utf-8 compiling for unicode charset on VS

### PR DESCRIPTION
**Pull request by dxzhan**
add compile option utf-8 for unicode on vs

### CODE REVIEW

**Comment by jackpoz** [comment]
does this fix any issue that you experienced ?

**Comment by Shauren** [comment]
I would love to know the reason for the CliRunnable change

**Comment by illfated** [comment]
interesting AppVeyor warnings (are the CP 65001 characters intended?):
```cpp
dep\g3dlite\include\G3D/Ray.h : warning C4828: The file contains a character starting at offset 0x29c that is illegal in the current source character set (codepage 65001).
dep\g3dlite\include\G3D/Ray.h : warning C4828: The file contains a character starting at offset 0x29d that is illegal in the current source character set (codepage 65001).
dep\g3dlite\include\G3D/Matrix3.h : warning C4828: The file contains a character starting at offset 0x2d4b that is illegal in the current source character set (codepage 65001).
dep\g3dlite\include\G3D/Matrix3.h : warning C4828: The file contains a character starting at offset 0x2d4c that is illegal in the current source character set (codepage 65001).
```

**Review by Shadowmaruz — APPROVED** [review]
This is a good commit.

I approve. 

That is all.

**Comment by dxzhan** [comment]
> interesting AppVeyor warnings (are the CP 65001 characters intended?):
> 
> ```c++
> dep\g3dlite\include\G3D/Ray.h : warning C4828: The file contains a character starting at offset 0x29c that is illegal in the current source character set (codepage 65001).
> dep\g3dlite\include\G3D/Ray.h : warning C4828: The file contains a character starting at offset 0x29d that is illegal in the current source character set (codepage 65001).
> dep\g3dlite\include\G3D/Matrix3.h : warning C4828: The file contains a character starting at offset 0x2d4b that is illegal in the current source character set (codepage 65001).
> dep\g3dlite\include\G3D/Matrix3.h : warning C4828: The file contains a character starting at offset 0x2d4c that is illegal in the current source character set (codepage 65001).
> ```

yes，i forgot commit it,sorry

**Comment by dxzhan** [comment]
> does this fix any issue that you experienced ?

fix the vs2017 chinese(mybe unicode charset) bad show

**Comment by dxzhan** [comment]
> I would love to know the reason for the CliRunnable change

fix the chinese show ????

**Comment by qyh214** [comment]
this pr is very well.It's good for like chinese show `?????`

but can you fix compile with some warnings?
QQ截图20200310163521

**Comment by illfated** [comment]
> this pr is very well.It's good for like chinese show `?????`
> 
> but can you fix compile with some warnings?
> QQ截图20200310163521
> 
> I have test and seems well in game and console

See my comment https://github.com/TrinityCore/TrinityCore/pull/24235#issuecomment-595006251 about interesting AppVeyor warnings.

```
dep\g3dlite\include\G3D/Ray.h     : warning C4828: The file contains a character starting at offset 0x29c  that is illegal in the current source character set (codepage 65001).
dep\g3dlite\include\G3D/Ray.h     : warning C4828: The file contains a character starting at offset 0x29d  that is illegal in the current source character set (codepage 65001).
dep\g3dlite\include\G3D/Matrix3.h : warning C4828: The file contains a character starting at offset 0x2d4b that is illegal in the current source character set (codepage 65001).
dep\g3dlite\include\G3D/Matrix3.h : warning C4828: The file contains a character starting at offset 0x2d4c that is illegal in the current source character set (codepage 65001).
```

**Comment by dxzhan** [comment]
> this pr is very well.It's good for like chinese show `?????`
> 
> but can you fix compile with some warnings?
> QQ截图20200310163521
> 
> I have test and seems well in game and console

convert file to utf-8 is build success

**Comment by dxzhan** [comment]
> > this pr is very well.It's good for like chinese show `?????`
> > but can you fix compile with some warnings?
> > QQ截图20200310163521
> > I have test and seems well in game and console
> 
> convert file to utf-8 is build success

i fix them and pushed

**Comment by dxzhan** [comment]
> > this pr is very well.It's good for like chinese show `?????`
> > but can you fix compile with some warnings?
> > QQ截图20200310163521
> > I have test and seems well in game and console
> 
> convert file to utf-8 is build success

i fix them and pushed

**Review by qyh214 — APPROVED** [review]

**Comment by illfated** [comment]
```cpp
C:\projects\trinitycore\src\tools\mmaps_generator\TerrainBuilder.cpp :
 error C2220: warning treated as error - no 'object' file generated
 [C:\projects\trinitycore\build\src\tools\mmaps_generator\mmaps_generator.vcxproj]
 
C:\projects\trinitycore\src\tools\mmaps_generator\TerrainBuilder.cpp :
 warning C4828: The file contains a character starting at offset 0x39c2 that is illegal in the current source character set (codepage 65001).
 [C:\projects\trinitycore\build\src\tools\mmaps_generator\mmaps_generator.vcxproj]

C:\projects\trinitycore\src\tools\mmaps_generator\TerrainBuilder.cpp :
 warning C4828: The file contains a character starting at offset 0x39ce that is illegal in the current source character set (codepage 65001).
 [C:\projects\trinitycore\build\src\tools\mmaps_generator\mmaps_generator.vcxproj]

C:\projects\trinitycore\src\tools\mmaps_generator\TerrainBuilder.cpp :
 warning C4828: The file contains a character starting at offset 0x39e8 that is illegal in the current source character set (codepage 65001).
 [C:\projects\trinitycore\build\src\tools\mmaps_generator\mmaps_generator.vcxproj]

C:\projects\trinitycore\src\tools\mmaps_generator\TerrainBuilder.cpp :
 warning C4828: The file contains a character starting at offset 0x39ed that is illegal in the current source character set (codepage 65001).
 [C:\projects\trinitycore\build\src\tools\mmaps_generator\mmaps_generator.vcxproj]
```

File: src/server/worldserver/CommandLine/CliRunnable.cpp [hunk 1]
```diff
@@ -91,7 +93,9 @@ void utf8print(void* /*arg*/, char const* str)
     if (!Utf8toWStr(str, strlen(str), wtemp_buf, wtemp_len))
         return;
 
-    wprintf(L"%s", wtemp_buf);
+    char temp_buf[6000];
+    CharToOemBuffW(&wtemp_buf[0], &temp_buf[0], wtemp_len+1);
```
Comment (jackpoz) [inline src/server/worldserver/CommandLine/CliRunnable.cpp]: could you post a test case we could use to test this ? did you investigate why wprintf doesn't work as expected in this case ?

File: dep/g3dlite/include/G3D/Ray.h [hunk 2]
```diff
@@ -36,7 +36,7 @@ class Ray {
     
     /** The following are for the "ray slope" optimization from
       "Fast Ray / Axis-Aligned Bounding Box Overlap Tests using Ray Slopes" 
-      by Martin Eisemann, Thorsten Grosch, Stefan M��ller and Marcus Magnor
+      by Martin Eisemann, Thorsten Grosch, Stefan M侟ller and Marcus Magnor
```
Comment (Shauren) [inline dep/g3dlite/include/G3D/Ray.h]: that really doesnt look right, a chinese character instead of a german one

Comment (illfated) [reply dep/g3dlite/include/G3D/Ray.h]: That is one of the 4 warnings in AppVeyor, if I am not mistaken.

Comment (illfated) [reply dep/g3dlite/include/G3D/Ray.h]: @dxzhan : Please revert the kanji characters back to their original Latin characters.

**Comment by qyh214** [comment]
@dxzhan revert e0c75f1189c8578178ad62a8b98438cd6d5ac982 and seems well but maybe CI will give some extra info.

@jackpoz I do not know many c++ but this commit fix console.
because If I send command in console like `.save all`
and I use Chinese and have set locale_Trinity_string
the message will show `???????` in console (the right message should be Chinese,for example)
QQ截图20200321154702

**Comment by Aokromes** [comment]
revert the last sync.

**Comment by jackpoz** [comment]
Please don't open a PR from 3.3.5 branch. Create a new branch instead, push your changes to that branch and then PR it. After that don't add any unrelated commits to that branch.

**Pull request closed without merging**